pkgs/development/web/publii/default.nix

```diff
@@ -0,0 +1,103 @@
+{ stdenv
+, lib
+, fetchurl
+, autoPatchelfHook
+, makeShellWrapper
+, wrapGAppsHook
+, alsa-lib
+, at-spi2-atk
+, at-spi2-core
+, atk
+, cairo
+, cups
+, dbus
+, expat
+, glib
+, gtk3
+, libsecret
+, mesa
+, nss
+, pango
+, udev
+, xdg-utils
+, xorg
+}:
+
+stdenv.mkDerivation rec {
+  pname = "Publii";
+  version = "0.42.0";
+
+  src = fetchurl {
+    url = "https://getpublii.com/download/Publii-${version}.deb";
+    hash = "sha256-ohwoKhLT75gOC4f6R1tjhSygNw5lR7kbWBG6qpcP368=";
+  };
+
+  dontConfigure = true;
+  dontBuild = true;
+  dontWrapGApps = true;
+
+  nativeBuildInputs = [
+    autoPatchelfHook
+    makeShellWrapper
+    wrapGAppsHook
+  ];
+
+  buildInputs = [
+    alsa-lib
+    at-spi2-atk
+    at-spi2-core
+    atk
+    cairo
+    cups
+    dbus
+    expat
+    glib
+    gtk3
+    libsecret
+    mesa
+    nss
+    pango
+    xorg.libX11
+    xorg.libxcb
+  ];
+
+  unpackPhase = ''
+    ar p $src data.tar.xz | tar xJ
+  '';
+
+  installPhase = ''
+    runHook preInstall
+
+    mkdir -p $out/bin
+
+    mv usr/share $out
+    substituteInPlace $out/share/applications/${pname}.desktop \
+      --replace 'Exec=/opt/Publii/Publii' 'Exec=${pname}'
+
+    mv opt $out
+    # ln -sf $out/opt/${pname}/${pname} $out/bin/${pname}
+
+    runHook postInstall
+  '';
+
+  preFixup = ''
+    # makeWrapper defaults to makeBinaryWrapper due to wrapGAppsHook
+    # but we need a shell wrapper specifically for xdg-fix. Make xdg-open overrideable at runtime.
+    # Electron is trying to open udev via dlopen()
+    # and for some reason that doesn't seem to be impacted from the rpath.
+    # Adding udev to LD_LIBRARY_PATH fixes that.
+    makeShellWrapper $out/opt/${pname}/${pname} $out/bin/${pname} \
+      "''${gappsWrapperArgs[@]}" \
+      --suffix PATH : ${lib.makeBinPath [ xdg-utils ]} \
+      --prefix LD_LIBRARY_PATH : ${lib.makeLibraryPath [ udev ]}
+  '';
+
+  meta = with lib; {
+    description = "Creating a website doesn't have to be complicated or expensive. With Publii, the most intuitive static site CMS, you can create a beautiful, safe, and privacy-friendly website quickly and easily; perfect for anyone who wants a fast, secure website in a flash.";
```

Comment:
generally `description` is a quick sentance/line, and `longDescription` is a paragraph